frontend/src/utils/utils.ts

```diff
@@ -49,8 +49,49 @@ export function zip<A, B>(a: A[], b: B[]): [A, B][] {
   return a.map((e, i) => [e, b[i]]);
 }
 
+// We show at most 5 characters. We empirically determined that 5 characters
+// can be fit into the IC50 column.
+// If we return a string that doesn't fit into the column for some reason,
+// we have fallback logic in CSS that limits the width and adds ellipsis.
 export function formatIC50(x: number): string {
-  return x.toFixed(3);
+  // We are carefully listing all cases here to think about each of them.
+  // Perhaps it can be optimized, e. g. using 'toPrecision', but I (@gromak)
+  // decided to follow the KISS principle.
+  if (x >= 1e5) {
```

Comment:
Maybe it's better to use switch or lookup map?
https://betterprogramming.pub/stop-putting-so-many-if-statements-in-your-javascript-3b65aaa4b86b

Reply:
How to use `switch` here? AFAIU switch works when you compare with some values for equality, but here I compare only for `>=`. Same for lookup map. I am new to JS/TS, so maybe I'm missing something, but it's not clear to me how to use `switch` here.

Reply:
switch:
```
values=[1000,  2000 ... 29000, 30000];

switch(true) {
  case (val < values[0]): /* do something */ break;
  case (val < values[1]): /* do something */ break;
  ...
  case (val < values[29]): /* do something */ break;
}
```

with lookup map.. maybe there are option to use expressions as keys.. but maybe I'm wrong and this is from other language

Reply:
Hm, ok, I'll give it a try.

Reply:
it's good now (and faster)